tests/integration/test_helloworld.py

```diff
@@ -13,10 +13,29 @@
 
 
 @pytest.mark.timeout(360)
-def test_helloworld(tmpdir):
+def test_helloworld_call(tmpdir):
     subprocess.check_call(['jina', 'hello-world', '--workdir', str(tmpdir)])
 
 
+@pytest.mark.timeout(360)
+def test_helloworld(tmpdir):
+    cmd = ['jina', 'hello-world', '--workdir', str(tmpdir)]
+    popen = subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                             universal_newlines=True)
+    is_hello_world = False
+    for stdout_line in iter(popen.stdout.readline, ""):
+        #  'cli = hello-world' is in stdoutput of hello-world script
+        #  this should be an indicator that hello-world script is executed
+        if 'cli = hello-world' in stdout_line:
+            is_hello_world = True
+            popen.terminate()
+            break
+    popen.stdout.close()
+    popen.wait()
+    assert is_hello_world, 'No cli = hello-world in stdoutput,' \
+                           'probably hello-world wasnt executed'
```

Comment:
Hi @bio-howard thanks for the test, I like it!
several points:
1. Can you rename `popen` to something like `p` or `proc`?
2. Can you use context manager, i.e. `with subprocess.Popen(..) as proc:` to ensure the proc being killed after running?
3. maybe rename `is_hellow_world` to `is_hello_world_in_stdout`?

Reply:
Thanks!
1. yes, proc is more meaningful
2. I will try to do it with context manager.
3. yes, it's more  accurate name